core-common/src/main/java/org/glassfish/jersey/message/internal/HeaderUtils.java

```diff
@@ -298,6 +299,24 @@ public static void checkHeaderChanges(final Map<String, String> headersSnapshot,
         }
     }
 
+    /**
+     * Compare two NewCookies having the same name. See documentation RFC.
+     *
+     * @param one    NewCookie to be compared.
+     * @param second NewCookie to be compared.
+     * @return the prefered NewCookie according to rules :
+     *              - the latest expiration date.
+     *              - if same expiration date, the longest path.
+     */
+    public static NewCookie getPreferedNewCookie(NewCookie one, NewCookie second) {
+
+        if (!one.getExpiry().equals(second.getExpiry())){
```

Comment:
expiry date can be NULL. Isn't it better to use maxAge value?

Reply:
There is a difference between expiry and maxAge. 
IE does not support maxAge, so most of the servers set expiry rather than maxAge. It is possible to set one or the other.
NewCookie is what comes from the server, and even though maxAge is recommended, expires seems to be more often. But it seems to be one or the other, so we should compare the later from both.

But yes, we should check for NULL, too.

Reply:
Well ,regarding check of expiry VS max-age it's good to check them both :) 

Regarding setting one or another attribute in request - actually server shall put both. For now I've found that the only browser which does not support max-age attribute is IE. 

Per RFC 6265 max-age attribute has precedence over expiry attribute if both are present in request. In my opinion this shall be respected while choosing preferred cookie.